repositories/pokemon_repo.go

```diff
@@ -0,0 +1,61 @@
+package repositories
+
+import (
+	"encoding/csv"
+	"fmt"
+	"golangBootcamp/m/models"
+	"os"
+	"strconv"
+
+	"github.com/spf13/viper"
+)
+
+type PokemonCSV interface {
+	ReadCsvFile(filePath string) ([][]string, error)
+}
+
+type PokemonRepo struct {
+	pokemonCSV PokemonCSV
+}
+
+func NewPokemonRepo(pokemonCsv PokemonCSV) PokemonRepo {
+	return PokemonRepo{pokemonCsv}
+}
+
+func (pr PokemonRepo) GetPokemonsFromCSV() ([]models.Pokemon, error) {
+	csvPokemons, error := pr.pokemonCSV.ReadCsvFile(viper.GetString("data.pokemon.file"))
+	if error != nil {
+		return nil, error
+	}
+	pokemons := []models.Pokemon{}
+	for _, csvPokemon := range csvPokemons {
+		intId, error := strconv.Atoi(csvPokemon[0])
+		if error != nil {
+			return nil, error
+		}
+		pokemons = append(pokemons, models.Pokemon{Id: intId, Name: csvPokemon[1]})
+	}
+	return pokemons, nil
+}
+
+func (pr PokemonRepo) WritePokemonCsvFile(pokemons []models.Pokemon) (bool, error) {
+	file, error := os.Create(viper.GetString("data.pokemon.file"))
```

Comment:
Just like you handle reading the csv file on another package, it would be good to move the logic that writes to csv files there as well. That way the logic will look consistent